tests/test_source.py

```diff
@@ -110,3 +110,37 @@ def test_source_multiple_photons_per_mode():
     s = Source(emission_probability=ep)
     svd = s.probability_distribution(2)
     assert_svd_close(svd, dict2svd({"|0>": (1-ep)**2, "|1>": ep*(1-ep)*2, "|2>": ep**2}))
+
+
+def test_source_sample():
+    from perceval.utils import BasicState
+
+    nb_samples = 200
+    source = Source(emission_probability=0.9, multiphoton_component=0.1, losses=0.1, indistinguishability=0.9)
+
+    for bs in [BasicState("|1,1>"), BasicState("|{_:0},{_:1}>")]: # with or without annotations
```

Comment:
I'm not sure using annotated input in this method makes sense (to be seen with the others)